### ISSUE TITLE
MQTT failing to connect causes lockup

### ISSUE THREAD

**Issue by glynhudson** [OpenEVSE/ESP8266_WiFi_v2.x#98]
If MQTT fails to connect (MQTT server is down) this causes the openevse to lock-up and web-page hangs at "loading"

```
OpenEVSE WiFI 1383250
Firmware: 2.3.0
Free: 31160
Connecting to SSID: Crumble
Connected, IP: 192.168.1.102
Server started
Free: 27368
MQTT Connecting to...emonpi

ng to...emonpi
MQTT failed: -2
MQTT Connecting to...emonpi
MQTT failed: -2
MQTT Connecting to...emonpi
MQTT failed: -2
MQTT Connecting to...emonpi
MQTT failed: -2
MQTT Connecting to...emonpi
MQTT failed: -2
MQTT Connecting to...emonpi
MQTT failed: -2
MQTT Connecting to...emonpi
```

**Comment by jeremypoulter** [comment]
This will largely be down to the MQTT library being blocking, should be fixed by implementing #30

**Comment by glynhudson** [comment]
Does not seem to be an issue anymore
